src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java

```diff
@@ -887,6 +920,84 @@ public void testGetImportedKeysDW() throws SQLException {
+            getColumnMetaDataClass.put(COLUMN_SIZE, Integer.class);
+            getColumnMetaDataClass.put(BUFFER_LENGTH, Object.class);        // Not used
+            getColumnMetaDataClass.put(DECIMAL_DIGITS, Integer.class);
+            getColumnMetaDataClass.put(NUM_PREC_RADIX, Integer.class);
+            getColumnMetaDataClass.put(NULLABLE, Integer.class);
+            getColumnMetaDataClass.put(REMARKS, String.class);
+            getColumnMetaDataClass.put(COLUMN_DEF, String.class);
+            getColumnMetaDataClass.put(SQL_DATA_TYPE, Integer.class);
+            getColumnMetaDataClass.put(SQL_DATETIME_SUB, Integer.class);
+            getColumnMetaDataClass.put(CHAR_OCTET_LENGTH, Integer.class);
+            getColumnMetaDataClass.put(ORDINAL_POSITION, Integer.class);
+            getColumnMetaDataClass.put(IS_NULLABLE, String.class);
+            getColumnMetaDataClass.put(SCOPE_CATALOG, String.class);
+            getColumnMetaDataClass.put(SCOPE_SCHEMA, String.class);
+            getColumnMetaDataClass.put(SCOPE_TABLE, String.class);
+            getColumnMetaDataClass.put(SOURCE_DATA_TYPE, Short.class);
+            getColumnMetaDataClass.put(IS_AUTOINCREMENT, String.class);
+            getColumnMetaDataClass.put(IS_GENERATEDCOLUMN, String.class);
+            getColumnMetaDataClass.put(SS_IS_SPARSE, Short.class);
+            getColumnMetaDataClass.put(SS_IS_COLUMN_SET, Short.class);
+            getColumnMetaDataClass.put(SS_UDT_CATALOG_NAME, String.class);
+            getColumnMetaDataClass.put(SS_UDT_SCHEMA_NAME, String.class);
+            getColumnMetaDataClass.put(SS_UDT_ASSEMBLY_TYPE_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_CATALOG_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_SCHEMA_NAME, String.class);
+            getColumnMetaDataClass.put(SS_XML_SCHEMACOLLECTION_NAME, String.class);
+    	}
+    	
+        try (Connection conn = getConnection()) {
+            ResultSetMetaData metadata = conn.getMetaData().getColumns(null, null, tableName, null).getMetaData();
+
+            for (int i = 1; i < metadata.getColumnCount(); i++) {
+                // Ensure that there is a data type for every metadata column
+                String columnLabel = metadata.getColumnLabel(i);
+                String columnClassName = metadata.getColumnClassName(i);
+                Class<?> columnClass = null;
+                
+                try 
+                {
+                    columnClass = Class.forName(columnClassName);
+                }
+                catch (ClassNotFoundException ex)
+                {
+                    MessageFormat form = new MessageFormat(
+                                TestResource.getResource("R_classLoaderNotFoundForColumnType"));
+                    Object[] msgArgs = {columnClassName, columnLabel};
+                    fail(form.format(msgArgs));
+                }
```

Comment:
I think we just need to assertTrue that the expectedClass.getName().equals(columnClassName) as per repro in the issue

Reply:
Changed the logic for checking the returned column data type matches the expected data type to simplify the logic. The new logic removes this Class Loader test. There is a new tests that actually compares the actual and expected class name for each column but does not use an assert. Instead a R_expectedClassDoesNotMatchActualClass message is used which includes the actual and expected data types as well as the column. This was done because if just an assert was used it would not show which column was generating the issue in the metadata result set.